tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +684,173 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm64")), File.Exists (arm64txt), "arm64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm")), File.Exists (armtxt), "arm.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-x64")), File.Exists (x64txt), "x64.txt");
+		}
+
+		void DeleteAssets (string project_path)
+		{
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+			try {
+				if (Directory.Exists (xcassetsDir))
+					Directory.Delete (xcassetsDir, true);
+			}
+			catch (Exception e){
+				Console.WriteLine($"Deleting Assets.xcassets failed: {e.Message}");
+			}
+		}
+
+		void CopyAssets (string project_path)
+		{
+			DirectoryInfo testingAssetsDir = new DirectoryInfo (Path.Combine (project_path, @"../../TestingAssets"));
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+
+			if (!testingAssetsDir.Exists)
+				throw new DirectoryNotFoundException ($"TestingAssets directory does not exist or could not be found: {testingAssetsDir}");
+
+			if (!Directory.Exists (xcassetsDir))
+				CopyDirectoryContents (testingAssetsDir, xcassetsDir);
+		}
+
+		void CopyDirectoryContents (DirectoryInfo dir, string DestDirectory) {
+			Directory.CreateDirectory (DestDirectory);
+			foreach (FileInfo file in dir.GetFiles ()) {
+				string tempPath = Path.Combine (DestDirectory, file.Name);
+				file.CopyTo (tempPath, false);
+			}
+			foreach (DirectoryInfo subDir in dir.GetDirectories ()){
+				CopyDirectoryContents (subDir, Path.Combine (DestDirectory, subDir.Name));
+			}
+		}
+
+		JsonDocument ProcessAssets (string assetsPath) {
+			try {
+				using (Process myProcess = new Process ()) {
+					myProcess.StartInfo.UseShellExecute = false;
+					myProcess.StartInfo.RedirectStandardOutput = true;
+					myProcess.StartInfo.Arguments = $"--sdk iphoneos assetutil --info {assetsPath}";
+					myProcess.StartInfo.FileName = "xcrun";
+					myProcess.StartInfo.CreateNoWindow = true;
+					myProcess.Start ();
+
+					string output = myProcess.StandardOutput.ReadToEnd ();
+					StreamReader reader = myProcess.StandardOutput;
+					myProcess.WaitForExit ();
+
+					return JsonDocument.Parse (output);
+				}
+			}
+			catch (Exception e) {
+				Console.WriteLine($"Couldn't process: {e.Message}");
+			}
+			return null;
+		}
+
+		HashSet<string> FindAssets (JsonDocument doc)
+		{
+			var jsonArray = doc.RootElement.EnumerateArray ();
+			var foundElements = new HashSet<string> ();
+
+			foreach (var item in jsonArray) {
+				var result = IsTarget (item);
+				if (result is not null)
+					foundElements.Add (result);
+			}
+			return foundElements;
+		}
+
+		string IsTarget (JsonElement item) {
+			if (item.TryGetProperty ("AssetType", out var assetType)) {
+				foreach (var target in XCAssetTargets) {
+					var result = IsTarget (item, assetType, target);
+					if (result is not null)
+						return result;
```

Comment:
return result? You already check in the caller if it is null, and null seems to be acceptable

Reply:
Ah I do this since that foreach right above that line will return null multiple times while it is searching for useful assets and we only want to add it to the foundElements hashset when it finds a useful one

Reply:
I only want to return the result here if it is not null since it will be going through the foreach loop and checking if the current JsonElement matches the XCAssetTarget we are looking at in that iteration. If not, I want it to continue through all the XCAssetTargets and only return when it finds it, or if it does not find it.